modules/internal/String.chpl

```diff
@@ -1276,19 +1287,28 @@ module String {
   iter string.items() : string {
     var localThis: string = this.localize();
 
-    var i = 0;
-    while i < localThis.buffLen {
-      const curPos = localThis.buff+i;
-      const (decodeRet, cp, nBytes) = decodeHelp(buff=localThis.buff,
-                                                   buffLen=localThis.buffLen,
-                                                   offset=i,
-                                                   allowEsc=true);
-      var (newBuf, newSize) = bufferCopyLocal(curPos, nBytes);
-      newBuf[nBytes] = 0;
+    if localThis.isASCII() {
+      for i in this.byteIndices {
+      var (newBuff, allocSize) = bufferCopy(buf=this.buff, off=i:int,
+                                            len=1, loc=this.locale_id);
+      yield chpl_createStringWithOwnedBufferNV(newBuff, 1, allocSize, 1);
```

Comment:
Could you check somehow (in a test or just by looking at the code) that the resulting string has a null terminator?

Reply:
This PR makes sure that we set the last byte to 0 and adds a test about it for this case.